File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/StyleHelper.java
package com.cedarsoft.commons.javafx;

import javax.annotation.Nonnull;

import javafx.scene.Node;

/**
 * Helper class with styles
 */
public class StyleHelper {
  private StyleHelper() {
  }

  @Nonnull
  public static <T extends Node> T withStyleClass(@Nonnull T node, @Nonnull String styleClass) {
    node.getStyleClass().add(styleClass);
    return node;
  }
}

File: commons/javafx/src/main/kotlin/com/cedarsoft/commons/javafx/StyleHelper.kt
package com.cedarsoft.commons.javafx

import com.cedarsoft.annotations.JavaFriendly
import javafx.scene.Node

/**
 * Helper class with styles
 */
object StyleHelper {
  @JavaFriendly
  @JvmStatic
  fun <T : Node> withStyleClass(node: T, styleClass: String): T {
    return node.withStyleClass(styleClass)
  }
}

fun <T : Node> T.withStyleClass(styleClass: String): T {
  this.styleClass.add(styleClass)
  return this
}
